### PR TITLE
 [ElasticDL] Set up dependencies and infra for ElasticDL tests

### PR DESCRIPTION
**Pull request by terrytangyuan**
Part of #664. This PR includes:
* Mount necessary paths (for docker, kubernetes, and minikube) when running e2e and maxcompute tests in the container. Use host network in the container.
* Install additional dependencies for ElasticDL, ElasticDL CLI, and build some necessary images for testing.
* Add `sql/python/test_magic_elasticdl.py` for testing SQLFlow magic command in ipython.
* Add `hasElasticDLCmd()` to check whether ElasticDL CLI has been installed.

Note that some of these is commented out until ElasticDL has been open sourced.

### CODE REVIEW

File: scripts/test_e2e.sh [hunk 1]
```diff
@@ -46,7 +46,11 @@ DATASOURCE="mysql://root:root@tcp(127.0.0.1:3306)/?maxAllowedPacket=0"
...
 
 sqlflowserver --datasource=${DATASOURCE} &
-# e2e test for standar SQL
+# e2e test for standard SQL
 SQLFLOW_SERVER=localhost:50051 ipython sql/python/test_magic.py
 # e2e test for xgboost train and prediciton SQL.
 SQLFLOW_SERVER=localhost:50051 ipython sql/python/test_magic_xgboost.py
+# TODO(terrytangyuan): Enable this when ElasticDL is open sourced
+# e2e test for ElasticDL SQL
+# export SQLFLOW_submitter=elasticdl
```
Comment (terrytangyuan) [inline scripts/test_e2e.sh:55]: A question here: this doesn't quite work yet. Somehow even though I specified `SQLFLOW_submitter=elasticdl` here the script `test_magic_elasticdl.py` still uses TensorFlow as the submitter. During my local testing, I had to hard-code executor to use elasticdl submitter to get this working.

**Comment by terrytangyuan** [comment]
Travis failed with 
```
--- FAIL: TestEnd2EndMaxCompute (18.82s)
4723    main_test.go:312: prepare test dataset failed: Post https://service.cn.maxcompute.aliyun.com/api/projects/gomaxcompute_driver_w7u/instances: net/http: TLS handshake timeout`
```
which should be unrelated to this PR.

File: scripts/test_e2e.sh (hunk 1 above)
Comment (tonyyang-svail) [reply scripts/test_e2e.sh:55]: One possible reason: the `sqlflowserver` is started on line 48, when `SQLFLOW_submitter` is still `""`

File: .travis.yml [hunk 2]
```diff
@@ -40,9 +40,9 @@ script:
...
   - docker run -d --name=hive sqlflow/gohive:dev python3 -m http.server 8899
   - docker run --rm -v $GOPATH:/go --net=container:hive --entrypoint bash -w /go/src/github.com/sql-machine-learning/sqlflow sqlflow:latest scripts/test_hive.sh
-  - docker run --rm -v $GOPATH:/go -w /go/src/github.com/sql-machine-learning/sqlflow sqlflow:latest bash scripts/test_e2e.sh
   - bash scripts/setup_k8s_env.sh
-  - docker run -e MAXCOMPUTE_AK=$MAXCOMPUTE_AK -e MAXCOMPUTE_SK=$MAXCOMPUTE_SK --rm -v $GOPATH:/go -w /go/src/github.com/sql-machine-learning/sqlflow sqlflow:latest bash scripts/test_maxcompute.sh
+  - docker run --rm --net=host -v /var/run/docker.sock:/var/run/docker.sock -v $HOME/.kube:/root/.kube -v /home/$USER/.minikube/:/home/$USER/.minikube/ -v $GOPATH:/go -w /go/src/github.com/sql-machine-learning/sqlflow sqlflow:latest bash scripts/test_e2e.sh
+  - docker run -e MAXCOMPUTE_AK=$MAXCOMPUTE_AK -e MAXCOMPUTE_SK=$MAXCOMPUTE_SK --rm --net=host -v /var/run/docker.sock:/var/run/docker.sock -v $HOME/.kube:/root/.kube -v /home/$USER/.minikube/:/home/$USER/.minikube/ -v $GOPATH:/go -w /go/src/github.com/sql-machine-learning/sqlflow sqlflow:latest bash scripts/test_maxcompute.sh
```
Comment (tonyyang-svail) [inline .travis.yml:45]: Why adding `--net=host -v /var/run/docker.sock:/var/run/docker.sock -v $HOME/.kube:/root/.kube -v /home/$USER/.minikube/:/home/$USER/.minikube/` to the `docker run` command of `test_maxcompute.sh`?

Comment (terrytangyuan) [reply .travis.yml:45]: These additional set of args are needed in order to run minikube + docker inside the container correctly, which will be required by ElasticDL CLI.

File: scripts/test_e2e.sh (hunk 1 above)
Comment (terrytangyuan) [reply scripts/test_e2e.sh:55]: I see. I’ll look into this later when I start uncommenting these lines after open source.

**Review by tonyyang-svail — APPROVED** [review]

**Pull request merged**